akka-persistence-typed/src/main/scala/akka/persistence/typed/EventSourcedSignal.scala

```diff
@@ -16,6 +16,7 @@ import akka.persistence.SnapshotSelectionCriteria
  */
 @DoNotInherit
 sealed trait EventSourcedSignal extends Signal
+sealed trait RetentionSignal extends EventSourcedSignal
```

Comment:
Do we need? No, however in tests it allows insight to what event occurred and with what value for deletes. Y/N?

Reply:
At least I hope it will, right now, the test is not liking this

Reply:
If you think about it from a user point of view, I'd expect that you want to act on ok vs failure and in case of failure you will want to look at the exception/sequence number, so there you have matching on the concrete types anyway. I think the motivation that there is a test that wants to see if any of these happens is not good enough, especially when you can just `TestProbe[AnyRef]()` and let it accept any type of message/signal.

If we decide to keep this there should be corresponding super types for other event sourced signals for consistency - recovery completed + recovery failed, snapshot completed + snapshot failed.

Reply:
probes can anyway not receive signals, so it must anyway be wrapped in a custom test message

Reply:
Removed